### PR TITLE
Add doc comment to hiding portions of code example

### LINKED ISSUE #50816: rustdoc does not clarify that examples need to be prepended with ///
See https://users.rust-lang.org/t/are-the-comments-about-using-in-rustdoc-outdated/17477 for details. Let me know if anything needs clarification.

### PR DESCRIPTION
**Pull request by mandeep**
fixes #50816 

Not sure if this is all that's needed, but I think it's a good start. One thing to note is that the code block is a text block where it could possibly be a rust block.

### CODE REVIEW

File: src/doc/rustdoc/src/documentation-tests.md [hunk 1]
```diff
@@ -79,8 +79,9 @@ from your example, but are important to make the tests work. Consider
 an example block that looks like this:
 
 ```text
-/// Some documentation.
-# fn foo() {}
+/// /// Some documentation.
+/// # fn foo() {} // this function will be hidden
+/// println!("Hello, World!");
```
Comment (GuillaumeGomez) [inline src/doc/rustdoc/src/documentation-tests.md]: Why is there a `println!`?

Comment (steveklabnik) [reply src/doc/rustdoc/src/documentation-tests.md]: yes, r=me with this line removed :)

**Review by steveklabnik — CHANGES_REQUESTED** [review]

**Comment by mandeep** [comment]
The foo function is there to show how to hide code. I wanted to add the println to show how normal code would look like in addition to hidden code.

**Comment by GuillaumeGomez** [comment]
Then you definitely need to update the next code example as well. But that seems really strange. Why would you put a doc comment on a hidden function? Doesn't make sense to me...

**Comment by mandeep** [comment]
The PR attempts to show users that one needs doc comments on all of their example code so that they don't write something like the following:

```rust
/// # Examples
/// Some documentation
# fn foo()
/// println("Hi");
```

**Comment by steveklabnik** [comment]
@mandeep that makes sense, but I'd prefer the examples to have actual syntax, then. You need curlies, for example.

**Comment by mandeep** [comment]
@steveklabnik Are curly braces needed somewhere besides the foo function?

**Comment by mandeep** [comment]
I discussed this on IRC and there may have been confusion with my example in my comment that contains code. I've updated the example to match the commit of the PR.

**Comment by pietroalbini** [comment]
Ping from triage! What's the status on this?

**Comment by mandeep** [comment]
@GuillaumeGomez maybe an example will help:

```rust
/// Find the barycentric coordinates of the given point with respect to the given triangle
///
/// # Examples
///
/// ```
/// let points =  vec![Vector3::new(0, 0, 0), Vector3::new(2, 2, 2), Vector3::new(0, 2, 2)]
/// let point = Point3::new(1.0, 1.0, 0.0);
/// let barycentric_coordinates: Point3<f64> = find_barycentric(&points, &point);
/// ```
///
fn find_barycentric(points: &Vec<Point3<f64>>, point: &Point3<f64>) -> Point3<f64> {
    let u = Vector3::new(points[2].x - points[0].x, points[1].x - points[0].x, points[0].x - point.x);
    let v = Vector3::new(points[2].y - points[0].y, points[1].y - points[0].y, points[0].y - point.y);

    let w = u.cross(&v);

    if (w.z).abs() < 1.0 {
        return Point3::new(-1.0, 1.0, 1.0);
    } else {
        return Point3::new(1.0 - (w.x + w.y) as f64 / w.z as f64, w.y as f64 / w.z as f64, w.x as f64 / w.z as f64);
    }

}
```

**Comment by GuillaumeGomez** [comment]
I still don't get it. :-/

**Comment by mandeep** [comment]
The use case is not immediately apparent, but since this omission is causing problems I think we need to add this commit. If you think more information is needed, let me know and I'd be happy to add more.

**Comment by GuillaumeGomez** [comment]
Ok, then let's just move forward.

@bors: r+ rollup

**Comment by jamesray1** [comment]
LGTM!

File: src/doc/rustdoc/src/documentation-tests.md (hunk 1 above)
Comment (jamesray1) [reply src/doc/rustdoc/src/documentation-tests.md]: LGTM!

**Pull request merged**